llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -421,8 +421,9 @@ static bool hasIrregularType(Type *Ty, const DataLayout &DL) {
   return DL.getTypeAllocSizeInBits(Ty) != DL.getTypeSizeInBits(Ty);
 }
 
-/// Returns "best known" trip count for the specified loop \p L as defined by
-/// the following procedure:
+/// Returns "best known" trip count, which is either a valid positive trip count
```

Comment:
Should  we specifically clarify that we return std::nullptr if there is a valid trip count but it doesn’t fit into unsigned? Otherwise it might be a bit surprising for cases where there is a valid known trip, it just doesn’t fit